python_modules/dagster/dagster/_core/definitions/metadata/metadata_value.py

```diff
@@ -1017,3 +1030,29 @@ class NullMetadataValue(NamedTuple("_NullMetadataValue", []), MetadataValue[None
     def value(self) -> None:
         """None: The wrapped null value."""
         return None
+
+
+@whitelist_for_serdes
+class CodeLocationStateMetadataValue(
+    NamedTuple("_CodeLocationStateMetadataValue", [("data", PublicAttr[Any])]), MetadataValue[Any]
+):
+    """Representation of some state data used to define the Definitions in a code location.
+
+    Args:
+        data (Any): Arbitrary JSON-serializable data used to define the Definitions in a
+            code location.
+    """
+
+    def __new__(cls, data: Any):
+        try:
+            json.dumps(data)
```

Comment:
hmmm shouldn't we be testing if it is packable by whitelist_for_serdes? Is json.dump a strict subset of that?

Reply:
Pretty sure `json.dump` is a subset. JSON handles str, int, float, bool, list, dict, None. All of that is handled by serdes.

Reply:
IMO we should just do an underlying `serdes` check since that is the true constraint.